torchvision/prototype/transforms/_misc.py

```diff
@@ -93,7 +98,12 @@ def _transform(
             )
 
         flat_tensor = inpt.reshape(-1, n) - self.mean_vector
-        transformed_tensor = torch.mm(flat_tensor, self.transformation_matrix)
+
+        transformation_matrix = self.transformation_matrix
+        if flat_tensor.dtype != transformation_matrix.dtype:
```

Comment:
wouldn't `.to` perform this check anyway? Are we avoiding a copy here?

Reply:
Yes, we can save one/two lines of code and do directly:
```python
transformation_matrix = self.transformation_matrix.to(flat_tensor.dtype)
```